compiler/rustc_mir_dataflow/src/impls/live_borrows.rs

```diff
@@ -0,0 +1,642 @@
+                        self.dep_graph.node(src_node_idx).data,
+                        node_idx,
+                        borrowed_local
+                    );
+
+                    self.dep_graph.add_edge(src_node_idx, node_idx, ());
+                }
+                None => {}
+            },
+            _ => {}
+        }
+
+        self.super_place(place, context, location)
+    }
+
+    #[instrument(skip(self), level = "debug")]
+    fn visit_operand(&mut self, operand: &Operand<'tcx>, location: Location) {
+        self.super_operand(operand, location)
+    }
+
+    #[instrument(skip(self), level = "debug")]
+    fn visit_terminator(&mut self, terminator: &Terminator<'tcx>, location: Location) {
+        match &terminator.kind {
+            TerminatorKind::Call { destination, func, args, .. } => {
+                let dest_ty = destination.ty(self.local_decls, self.tcx).ty;
+                debug!(?dest_ty);
+
+                let mut has_refs_or_pointers_visitor = MaybeHasRefsOrPointersVisitor::new();
+                dest_ty.visit_with(&mut has_refs_or_pointers_visitor);
+
+                let dest_might_include_refs_or_pointers =
+                    if has_refs_or_pointers_visitor.has_refs_or_pointers {
+                        true
+                    } else if has_refs_or_pointers_visitor.has_params {
+                        // if we pass in any references or raw pointers as arguments and the
+                        // return type includes type parameters, these type parameters could
+                        // refer to those refs/ptrs, so we have to be conservative here and possibly
+                        // include an edge.
+                        true
+                    } else {
+                        false
+                    };
+
+                let node_idx =
+                    self.dep_graph.add_node(NodeKind::LocalOrLocalWithRefs(destination.local));
+                debug!("added local {:?} to graph with idx {:?}", destination.local, node_idx);
+                self.locals_to_node_indexes.insert(destination.local, node_idx);
+
+                if dest_might_include_refs_or_pointers {
+                    self.current_local = Some(node_idx);
+                    debug!("self.current_local: {:?}", self.current_local);
+                }
+
+                self.visit_operand(func, location);
+                for arg in args {
+                    self.visit_operand(arg, location);
+                }
+
+                self.current_local = None;
+            }
+            TerminatorKind::Yield { resume_arg, value, .. } => {
+                let node_idx =
+                    self.dep_graph.add_node(NodeKind::LocalOrLocalWithRefs(resume_arg.local));
+                debug!("added local {:?} to graph with idx {:?}", resume_arg.local, node_idx);
+                self.locals_to_node_indexes.insert(resume_arg.local, node_idx);
+
+                self.super_operand(value, location);
```

Comment:
This should look more like the `Call` case: a yield is pretty much a call to an unknown function.